master/internal/api_user.go

```diff
@@ -683,3 +696,199 @@ func (a *apiServer) PostUserActivity(
+	}
+
+	tokenExpiration := user.DefaultTokenLifespan
+	if req.Lifespan != "" {
+		d, err := time.ParseDuration(req.Lifespan)
+		if err != nil || d < 0 {
+			return nil, status.Error(codes.InvalidArgument,
+				"Lifespan must be a Go-formatted duration string with a positive value")
+		}
+		tokenExpiration = d
+	}
+
+	token, err := user.DeleteAndCreateLongLivedToken(
+		ctx, curUser.ID, user.WithTokenExpiry(&tokenExpiration))
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.PostLongLivedTokenResponse{LongLivedToken: token}, nil
+}
+
+// POST User Long lived token takes user id and optional lifespan and overwrites an
+// access token for the given user.
+func (a *apiServer) PostUserLongLivedToken(
+	ctx context.Context, req *apiv1.PostUserLongLivedTokenRequest,
+) (*apiv1.PostUserLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	targetFullUser, err := getFullModelUser(ctx, model.UserID(req.UserId))
+	if err != nil {
+		return nil, err
+	}
+	targetUser := targetFullUser.ToUser()
+	if err = user.AuthZProvider.Get().CanCreateUsersToken(ctx, *curUser, targetUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenExpiration := user.DefaultTokenLifespan
+	if req.Lifespan != "" {
+		d, err := time.ParseDuration(req.Lifespan)
+		if err != nil || d < 0 {
+			return nil, status.Error(codes.InvalidArgument,
+				"Lifespan must be a Go-formatted duration string with a positive value")
+		}
+		tokenExpiration = d
+	}
+
+	token, err := user.DeleteAndCreateLongLivedToken(
+		ctx, targetFullUser.ID, user.WithTokenExpiry(&tokenExpiration))
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.PostUserLongLivedTokenResponse{LongLivedToken: token}, nil
+}
+
+// GET Long lived token returns token information for the logged in user.
+func (a *apiServer) GetLongLivedToken(
+	ctx context.Context, req *apiv1.GetLongLivedTokenRequest,
+) (*apiv1.GetLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanGetUsersOwnToken(ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenInfo, err := user.GetLongLivedTokenInfo(ctx, curUser.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiv1.GetLongLivedTokenResponse{LongLivedTokenInfo: tokenInfo.Proto()}, nil
+}
+
+// GET All Long lived token returns all long-lived token information from user_session db.
+func (a *apiServer) GetAllLongLivedToken(
+	ctx context.Context, req *apiv1.GetAllLongLivedTokenRequest,
+) (*apiv1.GetAllLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// targetFullUser, err := getFullModelUser(ctx, curUser.ID)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// targetUser := targetFullUser.ToUser()
```

Comment:
Are we wanting to keep this commented block?

Reply:
No, completely missed removing it. 
Will remove them

Reply:
Removed it!